src/libs/actions/Session.js

```diff
@@ -267,8 +269,14 @@ function setPassword(password, validateCode, accountID) {
 
             // This request can fail if the password is not complex enough
             Onyx.merge(ONYXKEYS.ACCOUNT, {error: response.message});
-        })
-        .finally(() => {
+        }).catch((errResponse) => {
+            const login = lodashGet(errResponse, 'data.email', null);
+            Onyx.merge(ONYXKEYS.ACCOUNT, {validated: false, validationCodeFailedMessage: 'resendValidationForm.validationCodeFailedMessage'});
```

Comment:
When we display the error in the UI then we can use the correct translation key. I think there's no reason to persist it in device storage.